test/modules/TwinTester/DesiredPropertiesValidator.cs

```diff
@@ -65,20 +56,21 @@ public async Task ValidateAsync()
             foreach (KeyValuePair<string, DateTime> desiredPropertyUpdate in desiredPropertiesUpdated)
             {
                 bool hasTwinUpdate = propertyUpdatesFromTwin.Contains(desiredPropertyUpdate.Key);
-                bool hasReceivedPotentialCallback = desiredPropertiesReceived.ContainsKey(desiredPropertyUpdate.Key) || !this.ShouldExpectDesiredPropertyCallback(this.twinState, desiredPropertyUpdate.Value);
+                bool hasCallbackReceived = desiredPropertiesReceived.ContainsKey(desiredPropertyUpdate.Key);
                 string status;
-                if (hasTwinUpdate && hasReceivedPotentialCallback)
+
+                if (hasTwinUpdate && (hasCallbackReceived || this.SkipCallbackValidationDueToEdgeHubRestart(this.twinTestState, desiredPropertyUpdate.Value)))
```

Comment:
I think it is better to consolidate these fields. I feel it makes the logic below (line 69, 73) awkward because hasCallbackReceived might not be needed.

The current implementation will work, but this seems error prone to me.